src/server/routes/collection.js

```diff
@@ -132,11 +134,16 @@ router.get('/:collectionId', auth.isAuthenticatedForCollectionView, async (req,
 			addedAt: item.added_at,
 			...await orm.func.entity.getEntity(orm, collection.entityType, item.bbid, relations)
 		}));
-		const entities = await Promise.all(entitiesPromise);
+		let entities = await Promise.all(entitiesPromise);
 		const isOwner = req.user && parseInt(collection.ownerId, 10) === parseInt(req.user?.id, 10);
 		const isCollaborator = req.user && collection.collaborators.filter(collaborator => collaborator.id === req.user.id).length;
 		const userId = req.user ? parseInt(req.user.id, 10) : null;
-
+		if (collection.entityType === 'Work') {
+			const workBBIDs = getListofBBIDs(entities);
```

Comment:
I don't know if we need a utility function for that, it should be as simple as 
```suggestion
			const workBBIDs = entities.map(entity => entity.bbid);
```

Reply:
you're right, my bad.